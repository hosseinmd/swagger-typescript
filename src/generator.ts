import {
  getPathParams,
  generateServiceName,
  getHeaderParams,
  getParametersInfo,
  getRefName,
  toPascalCase,
} from "./utils";
import type {
  SwaggerRequest,
  SwaggerJson,
  SwaggerResponse,
  SwaggerConfig,
  ApiAST,
  TypeAST,
  Schema,
  Parameter,
  ConstantsAST,
} from "./types";
import { generateApis } from "./generateApis";
import { generateTypes } from "./generateTypes";
import { generateConstants } from "./generateConstants";

function generator(input: SwaggerJson, config: SwaggerConfig): string {
  const apis: ApiAST[] = [];
  const types: TypeAST[] = [];
  let constantsCounter = 0;
  const constants: ConstantsAST[] = [];

  function getConstantName(value: string) {
    const constant = constants.find((_constant) => _constant.value === value);
    if (constant) {
      return constant.name;
    }

    const name = `_CONSTANT${constantsCounter++}`;

    constants.push({
      name,
      value,
    });

    return name;
  }

  try {
    Object.entries(input.paths).forEach(([endPoint, value]) => {
      Object.entries(value).forEach(
        ([method, options]: [string, SwaggerRequest]) => {
          const { operationId, security } = options;
          const parameters = options.parameters?.map((parameter) => {
            const { $ref } = parameter;
            if ($ref) {
              const name = $ref.replace("#/components/parameters/", "");
              return {
                ...input.components?.parameters?.[name],
                $ref,
                schema: { $ref } as Schema,
              } as Parameter;
            }
            return parameter;
          });

          const serviceName = generateServiceName(
            endPoint,
            method,
            operationId,
            config,
          );

          const pathParams = getPathParams(parameters);

          const {
            exist: queryParams,
            isNullable: isQueryParamsNullable,
            params: queryParameters,
          } = getParametersInfo(parameters, "query");
          let queryParamsTypeName: string | false = `${toPascalCase(
            serviceName,
          )}QueryParams`;

          queryParamsTypeName = queryParams && queryParamsTypeName;

          if (queryParamsTypeName) {
            types.push({
              name: queryParamsTypeName,
              schema: {
                type: "object",
                nullable: isQueryParamsNullable,
                properties: queryParameters?.reduce(
                  (prev, { name, schema, $ref, required, description }) => {
                    return {
                      ...prev,
                      [name]: {
                        ...($ref ? { $ref } : schema),
                        nullable: !required,
                        description,
                      } as Schema,
                    };
                  },
                  {},
                ),
              },
            });
          }

          const {
            params: headerParams,
            hasNullable: hasNullableHeaderParams,
          } = getHeaderParams(options.parameters, config);

          const requestBody = getBodyContent(options.requestBody);

          const contentType = Object.keys(
            options.requestBody?.content ||
              (options.requestBody?.$ref &&
                input.components?.requestBodies?.[
                  getRefName(options.requestBody.$ref as string)
                ]?.content) || {
                "application/json": null,
              },
          )[0];

          const accept = Object.keys(
            options.responses?.[200]?.content || {
              "application/json": null,
            },
          )[0];

          const responses = getBodyContent(options.responses?.[200]);

          let pathParamsRefString: string | undefined = pathParams.reduce(
            (prev, { name }) => `${prev}${name},`,
            "",
          );
          pathParamsRefString = pathParamsRefString
            ? `{${pathParamsRefString}}`
            : undefined;

          const additionalAxiosConfig = headerParams
            ? `{
              headers:{
                ...${getConstantName(`{
                  "Content-Type": "${contentType}",
                  Accept: "${accept}",

                }`)},
                ...headerParams,
              },
            }`
            : getConstantName(`{
              headers: {
                "Content-Type": "${contentType}",
                Accept: "${accept}",
              },
            }`);

          apis.push({
            summary: options.summary,
            deprecated: options.deprecated,
            serviceName,
            queryParamsTypeName,
            pathParams,
            requestBody,
            headerParams,
            isQueryParamsNullable,
            isHeaderParamsNullable: hasNullableHeaderParams,
            responses,
            pathParamsRefString,
            endPoint,
            method,
            security: security
              ? getConstantName(JSON.stringify(security))
              : "undefined",
            additionalAxiosConfig,
          });
        },
      );
    });

    if (input?.components?.schemas) {
      types.push(
        ...Object.entries(input.components.schemas).map(([name, schema]) => {
          return {
            name,
            schema,
          };
        }),
      );
    }

    if (input?.components?.parameters) {
      types.push(...Object.values(input.components.parameters));
    }
    if (input?.components?.requestBodies) {
      types.push(
        ...(Object.entries(input.components.requestBodies)
          .map(([name, _requestBody]) => {
            return {
              name: `RequestBody${name}`,
              schema: _requestBody.content?.["application/json"].schema,
            };
          })
          .filter((v) => v.schema) as any),
      );
    }

    let code = generateApis(apis);
    code += generateTypes(types);
    code += generateConstants(constants);

    return code;
  } catch (error) {
    console.error({ error });
    return "";
  }
}

function getBodyContent(responses?: SwaggerResponse): Schema | undefined {
  if (!responses) {
    return responses;
  }

  return responses.content
    ? Object.values(responses.content)[0].schema
    : responses.$ref
    ? ({
        $ref: responses.$ref,
      } as Schema)
    : undefined;
}

export { generator };
