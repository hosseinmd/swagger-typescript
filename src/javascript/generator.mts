import {
  getPathParams,
  generateServiceName,
  getHeaderParams,
  getParametersInfo,
  getRefName,
  toPascalCase,
} from "./utils.mjs";
import type {
  SwaggerRequest,
  SwaggerJson,
  SwaggerResponse,
  Config,
  ApiAST,
  TypeAST,
  Schema,
  Parameter,
  ConstantsAST,
  Method,
} from "../types.mjs";
import { generateApis } from "./generateApis.mjs";
import { generateTypes } from "./generateTypes.mjs";
import { generateConstants } from "./generateConstants.mjs";
import { generateHook } from "./generateHook.mjs";

function generator(
  input: SwaggerJson,
  config: Config,
): { code: string; hooks: string; type: string } {
  const apis: ApiAST[] = [];
  const types: TypeAST[] = [];
  let constantsCounter = 0;
  const constants: ConstantsAST[] = [];

  function getConstantName(value: string) {
    const constant = constants.find((_constant) => _constant.value === value);
    if (constant) {
      return constant.name;
    }

    const name = `_CONSTANT${constantsCounter++}`;

    constants.push({
      name,
      value,
    });

    return name;
  }

  try {
    Object.entries(input.paths).forEach(([endPoint, value]) => {
      const parametersExtended = value.parameters as Parameter[] | undefined;
      Object.entries(value).forEach(
        ([method, options]: [string, SwaggerRequest]) => {
          if (method === "parameters") {
            return;
          }

          const { operationId, security } = options;

          const allParameters =
            parametersExtended || options.parameters
              ? [...(parametersExtended || []), ...(options.parameters || [])]
              : undefined;

          const parameters = allParameters?.map<Parameter>((parameter) => {
            const { $ref } = parameter;
            if ($ref) {
              const name = $ref.replace("#/components/parameters/", "");
              return {
                ...input.components?.parameters?.[name]!,
                $ref,
                schema: { $ref } as Schema,
              };
            }
            return parameter;
          });

          const serviceName = generateServiceName(
            endPoint,
            method,
            operationId,
            config,
          );

          const pathParams = getPathParams(parameters);

          const {
            exist: isQueryParamsExist,
            isNullable: isQueryParamsNullable,
            params: queryParameters,
          } = getParametersInfo(parameters, "query");
          const queryParamsTypeName: string | false = isQueryParamsExist
            ? `${toPascalCase(serviceName)}QueryParams`
            : false;

          if (queryParamsTypeName) {
            types.push({
              name: queryParamsTypeName,
              schema: {
                type: "object",
                nullable: isQueryParamsNullable,
                properties: queryParameters?.reduce(
                  (
                    prev,
                    { name, schema, $ref, required: _required, description },
                  ) => {
                    return {
                      ...prev,
                      [name]: {
                        ...($ref ? { $ref } : schema),
                        nullable: !_required,
                        description,
                      } as Schema,
                    };
                  },
                  {},
                ),
              },
            });
          }

          const { params: headerParams, isNullable: hasNullableHeaderParams } =
            getHeaderParams(parameters, config);

          const requestBody = getBodyContent(options.requestBody);

          const contentType = Object.keys(
            options.requestBody?.content ||
              (options.requestBody?.$ref &&
                input.components?.requestBodies?.[
                  getRefName(options.requestBody.$ref as string)
                ]?.content) || {
                "application/json": null,
              },
          )[0] as ApiAST["contentType"];

          const accept = Object.keys(
            options.responses?.[200]?.content || {
              "application/json": null,
            },
          )[0];

          const responses = getBodyContent(options.responses?.[200]);

          let pathParamsRefString: string | undefined = pathParams.reduce(
            (prev, { name }) => `${prev}${name},`,
            "",
          );
          pathParamsRefString = pathParamsRefString
            ? `{${pathParamsRefString}}`
            : undefined;

          const additionalAxiosConfig = headerParams
            ? `{
              headers:{
                ...${getConstantName(`{
                  "Content-Type": "${contentType}",
                  Accept: "${accept}",

                }`)},
                ...headerParams,
              },
            }`
            : getConstantName(`{
              headers: {
                "Content-Type": "${contentType}",
                Accept: "${accept}",
              },
            }`);

          apis.push({
            contentType,
            summary: options.summary,
            deprecated: options.deprecated,
            serviceName,
            queryParamsTypeName,
            pathParams,
            requestBody,
            headerParams,
            isQueryParamsNullable,
            isHeaderParamsNullable: hasNullableHeaderParams,
            responses,
            pathParamsRefString,
            endPoint,
            method: method as Method,
            security: security
              ? getConstantName(JSON.stringify(security))
              : "undefined",
            additionalAxiosConfig,
            queryParameters,
          });
        },
      );
    });

    if (input?.components?.schemas) {
      types.push(
        ...Object.entries(input.components.schemas).map(([name, schema]) => {
          return {
            name,
            schema,
          };
        }),
      );
    }

    if (input?.components?.parameters) {
      types.push(
        ...Object.entries(input.components.parameters).map(([key, value]) => ({
          ...value,
          name: key,
        })),
      );
    }

    if (input?.components?.requestBodies) {
      types.push(
        ...(Object.entries(input.components.requestBodies)
          .map(([name, _requestBody]) => {
            return {
              name: `RequestBody${name}`,
              schema: Object.values(_requestBody.content || {})[0]?.schema,
              description: _requestBody.description,
            };
          })
          .filter((v) => v.schema) as any),
      );
    }

    let code = generateApis(apis, types, config);
    code += generateConstants(constants);
    const type = generateTypes(types, config);
    const hooks = config.reactHooks ? generateHook(apis, types, config) : "";

    return { code, hooks, type };
  } catch (error) {
    console.error({ error });
    return { code: "", hooks: "", type: "" };
  }
}

function getBodyContent(responses?: SwaggerResponse): Schema | undefined {
  if (!responses) {
    return responses;
  }

  return responses.content
    ? Object.values(responses.content)[0].schema
    : responses.$ref
    ? ({
        $ref: responses.$ref,
      } as Schema)
    : undefined;
}

export { generator };
