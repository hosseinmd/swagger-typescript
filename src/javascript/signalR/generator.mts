import type { TypeAST, Schema, Parameter, Config } from "../../types.mjs";
import { generateTypes } from "../generateTypes.mjs";
import { getDefineParam } from "../utils.mjs";

export interface HubJson {
  SignalrType: string;
  info: {
    title: string;
    termsOfService?: string;
    version: string;
  };
  hubs: {
    [serviceName: string]: {
      name: string;
      operations: { [x: string]: Operation };
      callbacks: { [x: string]: Operation };
    };
  };
  definitions: { [x: string]: Schema };
}

interface Operation {
  parameters: { [x: string]: Schema };
  description?: string;
  returntype?: Schema;
}

interface ParsedOperation {
  name: string;
  parameters: { [x: string]: Parameter };
  description?: string;
}

function signalRGenerator(json: HubJson, config: Config): string {
  const types: TypeAST[] = [];
  const hubs: {
    name: string;
    operations: ParsedOperation[];
    callbacks: ParsedOperation[];
  }[] = [];

  try {
    Object.values(json.hubs).map((hub) => {
      const operations: ParsedOperation[] = [];
      const callbacks: ParsedOperation[] = [];
      Object.entries(hub.operations).map(
        //@ts-ignore
        ([_name, operation]: [string, Operation]) => {
          operations.push({
            name: _name,
            parameters: operation.parameters as any,
            description: operation.description,
          });
        },
      );

      Object.entries(hub.callbacks).map(
        //@ts-ignore
        ([_name, operation]: [string, Operation]) => {
          callbacks.push({
            name: _name,
            parameters: operation.parameters as any,
            description: operation.description,
          });
        },
      );

      hubs.push({
        name: hub.name,
        operations,
        callbacks,
      });
    });

    if (json.definitions) {
      types.push(
        ...Object.entries(json.definitions as { [x: string]: Schema }).map(
          ([name, schema]) => {
            return {
              name,
              schema,
            };
          },
        ),
      );
    }
    let code = "";
    hubs.map(({ name: hubsName, operations, callbacks }) => {
      const operationEnumsName = `${hubsName}OperationsNames`;
      const operationEnums = operations
        .map(({ name: operationKey }) => `${operationKey} = "${operationKey}"`)
        .join(",\n");

      if (operationEnums) {
        code += `
          export enum ${operationEnumsName} {
            ${operationEnums}
            }\n`;

        code += `
          export type ${hubsName}Operations = {
            ${operations
              .map(
                ({ name, parameters }) =>
                  `[${operationEnumsName}.${name}]: (${Object.entries(
                    parameters,
                  ).map(([_name, schema]) =>
                    getDefineParam(
                      _name,
                      schema.required,
                      schema as unknown as Schema,
                      config,
                      schema.description,
                    ),
                  )}) => Promise<void>`,
              )
              .join(";\n")}
            };\n`;
      }

      const callbackEnumsName = `${hubsName}CallbacksNames`;
      const callbackEnums = callbacks
        .map(({ name: callbackKey }) => `${callbackKey} = "${callbackKey}"`)
        .join(",\n");

      if (callbackEnums) {
        code += `
          export enum ${callbackEnumsName} {
            ${callbackEnums}
            }\n`;

        code += `
          export type ${hubsName}Callbacks = {
            ${callbacks
              .map(
                ({ name, parameters }) =>
                  `[${callbackEnumsName}.${name}]: (${Object.entries(
                    parameters,
                  ).map(([_name, schema]) =>
                    getDefineParam(
                      _name,
                      schema.required,
                      schema as unknown as Schema,
                      config,
                      schema.description,
                    ),
                  )}) => void`,
              )
              .join(";\n")}
            };\n`;
      }

      code += `
      export interface ${hubsName} {
        ${
          callbackEnums
            ? `
          callbacksName: ${callbackEnumsName};
          callbacks: ${hubsName}Callbacks;
        `
            : ""
        }
        ${
          operationEnums
            ? `
          methodsName: ${operationEnumsName};
          methods: ${hubsName}Operations;
        `
            : ""
        }
      }
      `;
    });
    code += generateTypes(types, config);

    return code;
  } catch (error) {
    console.error({ error });
    return "";
  }
}

export { signalRGenerator };
