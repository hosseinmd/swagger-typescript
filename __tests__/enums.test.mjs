import { generator } from "../lib/javascript/generator.mjs";
import swaggerJson from "./swagger.json";

describe("generate enum as type", () => {
  const { code, hooks, type } = generator(swaggerJson, {
    generateEnumAsType: true,
  });

  it("generate Code", () => {
    expect(code).toMatchSnapshot();
  });
  it("generate hooks", () => {
    expect(hooks).toMatchSnapshot();
  });
  it("generate type", () => {
    expect(type).toMatchSnapshot();
  });
});

describe("generate enum", () => {
  const { code, hooks, type } = generator(swaggerJson, {});

  it("generate Code", () => {
    expect(code).toMatchSnapshot();
  });
  it("generate hooks", () => {
    expect(hooks).toMatchSnapshot();
  });
  it("generate type", () => {
    expect(type).toMatchSnapshot();
  });
});
