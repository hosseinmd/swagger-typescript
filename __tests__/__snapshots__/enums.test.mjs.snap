// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate enum as type generate Code 1`] = `
"
//@ts-nocheck
/**
* AUTO_GENERATED Do not change this file directly, use config.ts file instead
*
* @version 6
*/

import type { AxiosRequestConfig } from \\"axios\\";
import type { SwaggerResponse } from \\"./config\\";
import { Http } from \\"./httpRequest\\";
//@ts-ignore
import qs from \\"qs\\";
import type { GetSettlementV1PrivateSettlementGetQueryParams, SettlementRequestResultListResponseVM,}  from \\"./types\\"

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const __DEV__ = process.env.NODE_ENV !== \\"production\\";

// eslint-disable-next-line @typescript-eslint/no-unused-vars
function overrideConfig(
  config?: AxiosRequestConfig,
  configOverride?: AxiosRequestConfig,
): AxiosRequestConfig {
  return {
    ...config,
    ...configOverride,
    headers: {
      ...config?.headers,
      ...configOverride?.headers,
    },
  };
}

// eslint-disable-next-line @typescript-eslint/no-unused-vars
export function template(path: string, obj: { [x: string]: any } = {}) {
    Object.keys(obj).forEach((key) => {
      const re = new RegExp(\`{\${key}}\`, \\"i\\");
      path = path.replace(re, obj[key]);
    });

    return path;
}

// eslint-disable-next-line @typescript-eslint/no-unused-vars
function objToForm(requestBody: object) {
  const formData = new FormData();

  Object.entries(requestBody).forEach(([key, value]) => {
    value && formData.append(key, value);
  });

  return formData;
}

// eslint-disable-next-line @typescript-eslint/no-unused-vars
function objToUrlencoded(requestBody: object) {
  return qs.stringify(requestBody)
}

export const getSettlementV1PrivateSettlementGet = (
    queryParams?: GetSettlementV1PrivateSettlementGetQueryParams,configOverride?:AxiosRequestConfig
): Promise<SwaggerResponse< & (SettlementRequestResultListResponseVM)>> => {
  
  return Http.getRequest(
    getSettlementV1PrivateSettlementGet.key,
    queryParams,
    undefined,
    _CONSTANT1,
    overrideConfig(_CONSTANT0,
      configOverride,
    )
  )
}

/** Key is end point string without base url */
getSettlementV1PrivateSettlementGet.key = \\"/settlement/v1/private/settlement/get\\";
export const _CONSTANT0 = {
              headers: {
                \\"Content-Type\\": \\"application/json\\",
                Accept: \\"text/plain\\",
              },
            };export const _CONSTANT1 = [{\\"bearer\\":[]}];"
`;

exports[`generate enum as type generate hooks 1`] = `""`;

exports[`generate enum as type generate type 1`] = `
"
//@ts-nocheck
/**
* AUTO_GENERATED Do not change this file directly, use config.ts file instead
*
* @version 6
*/

        
        export type EnumWithoutName =0 | 1 | 2 | 10 | 30;
        
        
        export interface GetSettlementV1PrivateSettlementGetQueryParams {\\"EndDate\\"?: string;\\"IsAuto\\"?: boolean;
/**
 * 
 * - Format: decimal
 */
\\"MinimumAmount\\"?: number;
/**
 * 
 * - Format: int32
 */
\\"Skip\\"?: number;
/**
 * 
 * - Format: date-time
 */
\\"StartDate\\"?: string;\\"States\\"?:  & (SettlementRequestStatus)[];
/**
 * 
 * - Format: int64
 */
\\"UserWalletId\\"?: number;\\"status\\"?: SettlementRequestStatus;}
        
        
        export type NotificationLevel =\\"Unknown\\" | \\"Default\\" | \\"Success\\" | \\"Info\\" | \\"Warning\\" | \\"Danger\\";
        
        
        export interface SettlementRequestResultListResponseVM {
/**
 * 
 * - Format: int64
 */
\\"totalCount\\": number;\\"settlementRequests\\"?: SettlementRequestResultResponseVM[];}
        
        
        export interface SettlementRequestResultResponseVM {\\"automaticSettlement\\": boolean;
/**
 * 
 * - Format: decimal
 */
\\"commissionAmount\\": number;
/**
 * 
 * - Format: date-time
 */
\\"createDateUtc\\": string;
/**
 * 
 * - Format: guid
 */
\\"creatorUserId\\": string;
/**
 * 
 * - Format: int32
 */
\\"domainId\\": number;
/**
 * 
 * - Format: int64
 */
\\"id\\": number;\\"status\\": SettlementRequestStatus;
/**
 * 
 * - Format: int64
 */
\\"userWalletId\\": number;\\"accountNumber\\"?: string;}
        
        
        export type SettlementRequestStatus =\\"Pending\\" | \\"Paid\\";
        
        
/**
 * 
 * - Format: int32
 */

        export type Type =\\"0\\" | \\"-1\\";
        "
`;

exports[`generate enum generate Code 1`] = `
"
//@ts-nocheck
/**
* AUTO_GENERATED Do not change this file directly, use config.ts file instead
*
* @version 6
*/

import type { AxiosRequestConfig } from \\"axios\\";
import type { SwaggerResponse } from \\"./config\\";
import { Http } from \\"./httpRequest\\";
//@ts-ignore
import qs from \\"qs\\";
import type { GetSettlementV1PrivateSettlementGetQueryParams, SettlementRequestResultListResponseVM,}  from \\"./types\\"

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const __DEV__ = process.env.NODE_ENV !== \\"production\\";

// eslint-disable-next-line @typescript-eslint/no-unused-vars
function overrideConfig(
  config?: AxiosRequestConfig,
  configOverride?: AxiosRequestConfig,
): AxiosRequestConfig {
  return {
    ...config,
    ...configOverride,
    headers: {
      ...config?.headers,
      ...configOverride?.headers,
    },
  };
}

// eslint-disable-next-line @typescript-eslint/no-unused-vars
export function template(path: string, obj: { [x: string]: any } = {}) {
    Object.keys(obj).forEach((key) => {
      const re = new RegExp(\`{\${key}}\`, \\"i\\");
      path = path.replace(re, obj[key]);
    });

    return path;
}

// eslint-disable-next-line @typescript-eslint/no-unused-vars
function objToForm(requestBody: object) {
  const formData = new FormData();

  Object.entries(requestBody).forEach(([key, value]) => {
    value && formData.append(key, value);
  });

  return formData;
}

// eslint-disable-next-line @typescript-eslint/no-unused-vars
function objToUrlencoded(requestBody: object) {
  return qs.stringify(requestBody)
}

export const getSettlementV1PrivateSettlementGet = (
    queryParams?: GetSettlementV1PrivateSettlementGetQueryParams,configOverride?:AxiosRequestConfig
): Promise<SwaggerResponse< & (SettlementRequestResultListResponseVM)>> => {
  
  return Http.getRequest(
    getSettlementV1PrivateSettlementGet.key,
    queryParams,
    undefined,
    _CONSTANT1,
    overrideConfig(_CONSTANT0,
      configOverride,
    )
  )
}

/** Key is end point string without base url */
getSettlementV1PrivateSettlementGet.key = \\"/settlement/v1/private/settlement/get\\";
export const _CONSTANT0 = {
              headers: {
                \\"Content-Type\\": \\"application/json\\",
                Accept: \\"text/plain\\",
              },
            };export const _CONSTANT1 = [{\\"bearer\\":[]}];"
`;

exports[`generate enum generate hooks 1`] = `""`;

exports[`generate enum generate type 1`] = `
"
//@ts-nocheck
/**
* AUTO_GENERATED Do not change this file directly, use config.ts file instead
*
* @version 6
*/

        
        export enum EnumWithoutName {0=0,1=1,2=2,10=10,30=30}
        
        
        export interface GetSettlementV1PrivateSettlementGetQueryParams {\\"EndDate\\"?: string;\\"IsAuto\\"?: boolean;
/**
 * 
 * - Format: decimal
 */
\\"MinimumAmount\\"?: number;
/**
 * 
 * - Format: int32
 */
\\"Skip\\"?: number;
/**
 * 
 * - Format: date-time
 */
\\"StartDate\\"?: string;\\"States\\"?:  & (SettlementRequestStatus)[];
/**
 * 
 * - Format: int64
 */
\\"UserWalletId\\"?: number;\\"status\\"?: SettlementRequestStatus;}
        
        
        export enum NotificationLevel {Unknown=\\"Unknown\\",Default=\\"Default\\",Success=\\"Success\\",Info=\\"Info\\",Warning=\\"Warning\\",Danger=\\"Danger\\"}
        
        
        export interface SettlementRequestResultListResponseVM {
/**
 * 
 * - Format: int64
 */
\\"totalCount\\": number;\\"settlementRequests\\"?: SettlementRequestResultResponseVM[];}
        
        
        export interface SettlementRequestResultResponseVM {\\"automaticSettlement\\": boolean;
/**
 * 
 * - Format: decimal
 */
\\"commissionAmount\\": number;
/**
 * 
 * - Format: date-time
 */
\\"createDateUtc\\": string;
/**
 * 
 * - Format: guid
 */
\\"creatorUserId\\": string;
/**
 * 
 * - Format: int32
 */
\\"domainId\\": number;
/**
 * 
 * - Format: int64
 */
\\"id\\": number;\\"status\\": SettlementRequestStatus;
/**
 * 
 * - Format: int64
 */
\\"userWalletId\\": number;\\"accountNumber\\"?: string;}
        
        
        export enum SettlementRequestStatus {Pending=\\"Pending\\",Paid=\\"Paid\\"}
        
        
/**
 * 
 * - Format: int32
 */

        export enum Type {\\"0\\"=\\"0\\",\\"-1\\"=\\"-1\\"}
        "
`;
